packages/caver-account/src/accountKey/weightedMultiSigOptions.js

```diff
@@ -0,0 +1,117 @@
+ * @class
+ */
+class WeightedMultiSigOptions {
+    /**
+     * Creates an instance of WeightedMultiSigOptions.
+     * @param {object} options - An object which defines 'threshold' and 'weights'.
+     * @return {WeightedMultiSigOptions}
+     */
+    static fromObject(options) {
+        // To support the previous options format(threshold and weight)
+        if (options.weight && !options.weights) {
+            options.weights = options.weight
+            delete options.weight
+        }
+
+        // Returns empty WeightedMultiSigOptions if options not define threshold and weights like '{}'
+        if (options.threshold === undefined && options.weights === undefined) return new WeightedMultiSigOptions()
+
+        if (options.threshold === undefined || options.weights === undefined)
+            throw new Error(`Invalid object for creating WeightedMultiSigOptions. 'threshold' and 'weights' should be defined.`)
+
+        return new WeightedMultiSigOptions(options.threshold, options.weights)
+    }
+
+    /**
+     * Creates an instance of WeightedMultiSigOptions.
+     * @param {string} threshold - a threshold
+     * @param {Array.<number>} weights - an array of weight of key
+     */
+    constructor(threshold, weights) {
+        if ((threshold !== undefined && weights === undefined) || (threshold === undefined && weights !== undefined)) {
+            throw new Error(`For creating an WeightedMultiSigOptions, 'threshold' and 'weights' should be defined.`)
+        }
+        if (threshold !== undefined) this.threshold = threshold
+        if (weights !== undefined) this.weights = weights
+    }
+
+    /**
+     * @type {number}
+     */
+    get threshold() {
+        return this._threshold
+    }
+
+    set threshold(th) {
+        if (!_.isNumber(th)) throw new Error(`The threshold must be number type.`)
+
+        if (!validateWeightedSum(th, this.weights))
+            throw new Error('Invalid options for AccountKeyWeightedMultiSig: The sum of weights is less than the threshold.')
+
+        this._threshold = th
+    }
+
+    /**
+     * @type {Array.<number>}
+     */
+    get weights() {
+        return this._weights
+    }
+
+    set weights(weightArr) {
+        if (!_.isArray(weightArr)) throw new Error(`weight should be an array that stores the weight of each public key.`)
+        for (const w of weightArr) {
+            if (!_.isNumber(w)) throw new Error(`The weight of each key must be number type.`)
+        }
+
+        if (!validateWeightedSum(this.threshold, weightArr))
+            throw new Error('Invalid options for AccountKeyWeightedMultiSig: The sum of weights is less than the threshold.')
+
+        this._weights = weightArr
+    }
+
+    /**
+     * Returns 'true' if WeightedMultiSigOptions is empty.
+     * @return {Boolean}
+     */
+    isEmpty() {
+        return this.threshold === undefined && this.weights === undefined
+    }
+}
+
+function validateWeightedSum(threshold, weights) {
```

Comment:
Don't we need to have a description of this function?

Reply:
I added description for this function ! :)